Tile borders settings, autosave result, seed randomisation

1ort opened the pull request:
Added a setting for the width and color of the tile frame for greater aesthetics, the final result is now saved in the txt2img-grids folder.

Known issue: border width is not always pixel perfect due to tiles overlapping each other with a slight overlap.

I don't know what to do with the border around the final grid. At the moment, due to the way the frame is drawn, it is approximately twice as thin as the inner

1ort committed:
Tile borders settings and autosave result

Added a setting for the width and color of the tile frame for greater aesthetics, the final result is now saved in the txt2img-grids folder

deepserket commented on img2tiles.py:
```diff
@@ -1,14 +1,14 @@
+from email.mime import base
```
Can you explain why `email.mime.base` is imported?

1ort replied:
Lol, no idea at what point it ended up in my code. I'll fix it now, thanks for noticing

1ort committed:
Minor import fix + seed randomisation

Removed unused unnecessary import
Added a checkbox responsible for seed randomization for each tile

arcanite24 commented:
Looking good! @1ort , let me know when ready for review

1ort commented:
A fixed seed in some cases gives out a strong image repetitiveness, especially if the tiles of the original image are similar in color.

The "Use -1 for seeds" checkbox avoids this. Examples below: 
![test](https://user-images.githubusercontent.com/83316072/198346290-944bb9cc-29ad-4a43-862b-304f533d0dad.png)
source, 

![grid-0186](https://user-images.githubusercontent.com/83316072/198346407-ac132522-e7ea-48f8-837f-c987f9ac8eb9.png)
checkbox off, 

![grid-0185](https://user-images.githubusercontent.com/83316072/198346463-7aa60573-e1ec-44d1-b267-3b33746b329e.png)
checkbox on

1ort commented:
@arcanite24
That's all the code changes I'd like to make for today.
I tested all the introduced features and did not find problems with the script, except for the minor ones related to the width of the frames, which I described above
Thanks

upd. Please let me know if I should update the README with new features and examples or if you can handle it yourself

arcanite24 commented:
Great work @1ort !
README changes would be awesome describing your changes and updating the examples

1ort commented:
I was forced to close this pull request because the script doesn't work as it should in most of my tests. On any pictures except 512x and on any tile side sizes except 1x1.

I will be rewriting the script from scratch, using SD_upscale as the basis, so all my changes at the moment do not make sense

JLansey commented:
I've been playing around with this - and think it would be useful to be able to manually set the seeds for each image. This would be a way for you to keep variations that you like (without using a separate photo editor). Not sure if you have a better way to do this - or a photo editor that makes this easier.

SMUsamaShah commented:
@1ort Sorry for off-topic question but what was your prompt for above image.

BTW, saw your PR after submitting mine which simply only increments the seed for each tile.
